### PR TITLE
Add an automatic Github pages publish step for the action.

### PR DESCRIPTION
**Pull request by yilinwei**
Hi,

This adds an automatic publish step for the action since it's quite common to do - it works, insofar as it pushes the HTML into the correct branch et. cetera. Many thanks to @lexi-lambda for the original Travis CI script which I basically copied.

There is a slight problem with it so far - the links don't seem to point to the right location - what was the reason for the local linking?

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -30,3 +30,33 @@ fi
...
+	raco scribble +m --htmls \
+	     --redirect-main http://pkg-build.racket-lang.org/doc/ \
+	     --dest ./docs \
+	     "${INPUT_DOC_DIRECTORY}/scribblings/${INPUT_SCRIBBLE_FILE}.scribl"
+	# Here we create a /new/ repository with no history where the scribble docs are generated.
+	cd docs || exit 1
+	# Sometimes the scribble command will create a subfolder, but Github doesn't like it.
+	# This is a noop in the case that there isn't one.
+	cd $(find . -maxdepth 1 -type d | tail -n1) || exit 1
+        git config --global user.email $(git show -s --format=format:%ae)
```
Comment (yilinwei) [inline entrypoint.sh:51]: Literally do not understand why this isn't showing up as aligned.

**Comment by jackfirth** [comment]
I actually _prefer_ to just link to the package catalog docs. In my experience, trying to set up github pages documentation is very fiddly and prone to weird edge cases and glitches. Case in point: I have absolutely no idea how to fix that local linking issue. So I would rather avoid dealing with the ongoing maintenance work of supporting github pages docs (not to mention the added build time costs) and instead just rely on the official package catalog for docs.

Did you want to use this template for a package you're not going to publish on the package catalog?

**Comment by yilinwei** [comment]
That makes sense - I'll close the PR.

No - I don't intend on doing that at the moment; it was for two purposes primarily,

1. For an up-to-date link for the current master, regardless of when the package catalog would be crawling it
2. As a double-check to make sure that it would render correctly in a clean install,

I worked out the issue with the link - I had used the wrong flag `redirect-main` instead of [redirect](https://github.com/yilinwei/otp/blob/6ebd0ff0aae374e9379bce2e7a0456f7941fa485/docs.sh#L14).

**Comment by jackfirth** [comment]
Thanks for opening the pull request nonetheless, it's handy to have this to refer to if I ever need to figure out how to do this for a project.

**Pull request closed without merging**